staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -651,6 +665,39 @@ func (p *sharedProcessor) addListenerLocked(listener *processorListener) {
 	p.syncingListeners = append(p.syncingListeners, listener)
 }
 
+func (p *sharedProcessor) removeListener(index int) {
+	handler := p.listeners[index].handler
+	p.listeners = append(p.listeners[:index], p.listeners[index+1:]...)
+	for i := len(p.syncingListeners) - 1; i >= 0; i-- {
+		if p.syncingListeners[i].handler == handler {
+			p.syncingListeners = append(p.syncingListeners[:i], p.syncingListeners[i+1:]...)
+		}
+	}
+}
+
+func (p *sharedProcessor) removeEventHandler(handler ResourceEventHandler) bool {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	for i, listener := range p.listeners {
+		// This will panic if the handler is not comparable
```

Comment:
this is shared informer, possibly used by many clients, panic would affect all clients even though only one client misused it. what about returning an err?

Reply:
I will address in separate PR.

Reply:
Done. See https://github.com/kubernetes/kubernetes/pull/98536/files#diff-10a340e966883e5205df63ad90ba6827f38f8a8d4b8614152c0baae9813a56c6R637